state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1508,17 +1500,17 @@ pub mod pallet {
 			if total_fee_bps > MAX_BASIS_POINTS {
 				FeeTaken { remaining_amount: stable_amount, fee: 0 }
 			} else {
-				let total_fee = broker_fees.iter().fold(
-					0u128,
-					|fee_accumulator, Beneficiary { account, bps }| {
+				let total_fee = broker_fees
+					.iter()
+					.filter(|Beneficiary { account: _, bps }| *bps > 0)
```

Comment:
nit, a bit more succinct: 

```suggestion
					.filter(|Beneficiary { bps, .. }| *bps > 0)
```